### PR TITLE
Ashwin

### PR DESCRIPTION
**Pull request by ash-02**
created routes for reviews and coupons

### CODE REVIEW

File: Files/coupons/utils.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
+from unittest import result
```
Comment (akhilsrinivasp) [inline Files/coupons/utils.py:1]: unnecessary imports can be removed ig

File: Files/coupons/utils.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+def retrieve_coupon_id(coupon_id):
+    coupon_details = db.session.query(Coupons).filter(Coupons.coupon_id==coupon_id).first()
+    if not coupon_details:
+        return None
+    coupons_schema=CouponsSchema()
+    output = coupons_schema.dump(coupon_details)
+    return {"result":output}
+
+def add_coupon(discount, limit, minimum_cart_value):
+    coupon_code = generate_coupon_code()
```
Comment (AviralJ58) [inline Files/coupons/utils.py:32]: Coupon code must also be taken as input. For example there can be a coupon "POW100" offering flat Rs. 100 discount. Or something related to bank offers and stuff. Code shouldn't be random.

File: Files/models.py [hunk 3]
```diff
@@ -136,8 +136,8 @@ class Reviews (db.Model):
     user_id = db.Column(db.Integer, nullable=False)
     rating = db.Column(db.Integer, nullable=False)
     review = db.Column(db.String(80), nullable=False)
```
Comment (AviralJ58) [inline Files/models.py:138]: nullable=True as some users can just rate and not leave a review.

File: Files/models.py [hunk 4]
```diff
@@ -136,8 +136,8 @@ class Reviews (db.Model):
     user_id = db.Column(db.Integer, nullable=False)
     rating = db.Column(db.Integer, nullable=False)
```
Comment (AviralJ58) [inline Files/models.py:137]: Data type float to allow rating like 4.5

**Review by AviralJ58 — CHANGES_REQUESTED** [review]
Check out for comments. Also common.db is conflicting. Try deleting it and push. If it doesn't work then temporarily add the db from the main branch and push with that to solve the conflict.

**Pull request closed without merging**